staging/src/k8s.io/apiserver/pkg/endpoints/installer.go

```diff
@@ -947,7 +938,7 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 				}
 				handler := metrics.InstrumentRouteFunc(action.Verb, group, version, resource, subresource, requestScope, metrics.APIServerComponent, deprecated, removedRelease, restfulConnectResource(connecter, reqScope, admit, path, isSubresource))
 				handler = utilwarning.AddWarningsHandler(handler, warnings)
-				route := ws.Method(method).Path(action.Path).
+				route = ws.Method(method).Path(action.Path).
```

Comment:
I do not understand how it is correct to remove the `routes` array do to this line. This case adds a new route for each `method` in `connector.ConnectMethods()`. 

So the original code could end up with multiple `routes`, but your change only allows the last one? Do you have a guarantee that `connecter.ConnectMethods` is always one element? Otherwise, I do not see how this is a correct change to make.